File: src/main/java/my/maryvk/maryvkweb/seeker/FriendsAndFollowersMarySeekerFactory.java
package my.maryvk.maryvkweb.seeker;

import my.maryvk.maryvkweb.service.RelationService;
import my.maryvk.maryvkweb.service.VkService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class FriendsAndFollowersMarySeekerFactory implements MarySeekerFactory {

    private final VkService vk;
    private final RelationService relationService;

    @Autowired
    public FriendsAndFollowersMarySeekerFactory(VkService vk, RelationService relationService) {
        this.vk = vk;
        this.relationService = relationService;
    }

    public MarySeeker create(int userId) {
        return new FriendsAndFollowersMarySeeker(vk, userId, relationService);
    }

}

File: src/main/java/my/maryvk/maryvkweb/seeker/FriendsAndFollowersMarySeekerFactory.kt
package my.maryvk.maryvkweb.seeker

import my.maryvk.maryvkweb.service.RelationService
import my.maryvk.maryvkweb.service.VkService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class FriendsAndFollowersMarySeekerFactory(
        private val vk: VkService,
        private val relationService: RelationService
) : MarySeekerFactory {
    override fun create(userId: Int): MarySeeker {
        return FriendsAndFollowersMarySeeker(vk = vk, userId = userId, relationService = relationService)
    }
}